Main Idea:  The start time of a ride is dependent on the age of the rider.  
We will use Millenials (users born between 1980 and 2000) and non Millenials (users born outside of 1980-2000) as the age groups for this test.
We will be using the data from January 2015 and limiting our sample size to "Subscribers" so we can use their date of birth to create a subset. Date of birth is not given for "Customers"

Null Hypothesis: Age Group (Millenials and Non-Millenials) and late night riding (10PM-3AM) are independent 

Alternative Hypothesis: Age Group (Millenials and Non-Millenials) and late night riding (10PM-3AM) are dependent 

Confidence Level = 0.05

In [103]:
import os
import csv
import pylab as pl
import pandas as pd
import numpy as np
%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [104]:
df = pd.read_csv('201501-citibike-tripdata.csv')

In [105]:
df['starthour'] = pd.DatetimeIndex(df['starttime']).hour
df['STMillenial'] = df['starthour'][(df['usertype']=='Subscriber') & ((df['birth year']>=1980) & (df['birth year'] <=2000))]
df['STNotMillenial'] = df['starthour'][(df['usertype']=='Subscriber') & ((df['birth year']<1980) | (df['birth year'] >2000))]
#exclude all customers
df = df[df.usertype!= 'Customer']
#df.head

<bound method DataFrame.head of         tripduration        starttime         stoptime  start station id  \
0               1346    1/1/2015 0:01    1/1/2015 0:24               455   
1                363    1/1/2015 0:02    1/1/2015 0:08               434   
2                346    1/1/2015 0:04    1/1/2015 0:10               491   
3                182    1/1/2015 0:04    1/1/2015 0:07               384   
4                969    1/1/2015 0:05    1/1/2015 0:21               474   
5                496    1/1/2015 0:07    1/1/2015 0:15               512   
6                152    1/1/2015 0:07    1/1/2015 0:09               498   
7               1183    1/1/2015 0:08    1/1/2015 0:28               405   
8                846    1/1/2015 0:09    1/1/2015 0:23               450   
9                576    1/1/2015 0:10    1/1/2015 0:20               160   
10               540    1/1/2015 0:10    1/1/2015 0:19               291   
11               419    1/1/2015 0:11    1/1/2015 0:18  

In [106]:
#Late Night Data Frame
df2 = df.query('starthour < 3 or starthour >= 22')

#Not Late Night Data Frame
df3 = df.query('starthour>=3 and starthour < 22')

Chi Squared Test should be used for this data

In [107]:
#Total Subscribers
totsubscribers = float(df['starttime'].count())
#Total Late Night Subscribers
totLN = float(df2['starttime'].count())
#Total Not Late Night Subscribers
totNLN = float(df3['starttime'].count())
#total Millenials 
totMil = float(df['STMillenial'].count())
#total NonMillenials
totNonMil = float(df['STNotMillenial'].count())
#total Millenials Late Night
totMilLN = float(df2['STMillenial'].count())
#total NonMillenials Late Night
totNonMilLN = float(df2['STNotMillenial'].count())
#total Millenials Not Late Night
totMilNLN = float(df3['STMillenial'].count())
#total NonMillenials Not Late Night
totNonMilNLN = float(df3['STNotMillenial'].count())


In [108]:
expected = float((totLN*totNLN*totMil*totNonMil))
sample_values =  [[totNonMilLN, totMilLN], [totNonMilNLN, totMilNLN]]


#make a chi squared statistic function
chistat = lambda N, values, expect : float(N*((values[0][0]*values[1][1]-values[0][1]*values[1][0])**2)/(expect))


print chistat(totsubscribers, sample_values, expected)


860.362058299


We found a chi squared statistic of 860.36.
This chi statistic far exceeds 3.84 and we are able to reject our null hypothesis with confidence levl of .05 (alpha)